### DISCUSSION TITLE
DNSDIST webserver curl Empty reply

### DISCUSSION THREAD

**Discussion by walidwake** [PowerDNS/pdns#13541] [Q&A]
Hi all, i'm new to powerdns/dnsdist, and i  hope i'm asking my question in the right place.

I have instaled dnsdist and i'm trying to retieve metrics via the websever.

My config file is very simple for testing and contains only webserver config to listen on the 10.250.2.26:8083

`webserver("10.250.2.26:8083", "1234", "1234")`

When i try to curl to get the metrics via curl i get empty reply from server 

```
# curl -H 'X-API-Key: 1234' http://10.250.2.26:8083/metrics
curl: (52) Empty reply from server
```

but when i configure the webserver to listen on the 127.0.0.1:8083 i get a proper response

`webserver("127.0.0.1:8089", "1234", "1234")`

```
# curl -H 'X-API-Key: 1234' http://127.0.0.1:8083/metrics
# HELP dnsdist_responses Number of responses received from backends
# TYPE dnsdist_responses counter
dnsdist_responses 0
# HELP dnsdist_servfail_responses Number of SERVFAIL answers received from backends
# TYPE dnsdist_servfail_responses counter
dnsdist_servfail_responses 0
# HELP dnsdist_queries Number of received queries
# TYPE dnsdist_queries counter
dnsdist_queries 0
# HELP dnsdist_frontend_nxdomain Number of NXDomain answers sent to clients
# TYPE dnsdist_frontend_nxdomain counter
dnsdist_frontend_nxdomain 0

```

any idea on why it behaves like this ?

Thanks, regards.

**Comment by rgacogne** [comment]
The default ACL only allows connections from `127.0.0.1` and `::1`, see https://dnsdist.org/guides/webserver.html

**Comment by walidwake** [comment]
the command for whatever reason wasn't working with the config specified in the documentation, dnsdist coudn't start because of lua language exception.

```
webserver("127.0.0.1:8083")
setWebserverConfig({password="supersecretpassword", apiKey="supersecretAPIkey"})
```

but with this config it worked

```
webserver("10.250.2.27:8089", "1234", "1234")
setWebserverConfig({acl="10.255.0.0/16 10.250.0.0/16",password="1234"})
```

Thank a lot for support.
